### PR TITLE
Add description about Cargo js feature for WebAssembly section

### PR DESCRIPTION
**Pull request by Kazhuu**
Documentation improvement regarding https://github.com/rust-random/getrandom/issues/267.

### CODE REVIEW

**Comment by newpavlov** [comment]
You should add a clarification that the `js` feature SHOULD be enabled only in "application" crates. For library tests and benchmarks the feature should be enabled only for tests and benchmarks, not for the whole library.

**Comment by Kazhuu** [comment]
Okay I updated the doc specifically stating what you mentioned. I'm quite new to Rust in general so had to do some digging with this. Is it possible to enable specific feature when the crate type is binary for example? If yes, then example of this could be added to the docs.

For one of my own projects I've enabled js feature like this:
```TOML
[dependencies.getrandom]
features = ["js"]
```

**Comment by josephlr** [comment]
@Kazhuu I've made some updates to the wording here. Hopefully this will help emphasize that  this feature should generally be enabled on non-library crates (binary, benchmark, or test crates). I also removed the "Indirect Dependencies" Section as its now redundant.

I'll give @newpavlov a chance to review this. Note that the CI failure is caused by an unrelated nightly bug, see:
  - https://github.com/rust-osdev/uefi-rs/issues/494
  - https://github.com/rust-lang/rust/issues/101071
  - https://github.com/rust-lang/rust/pull/101088

**Review by josephlr — APPROVED** [review]

**Comment by Kazhuu** [comment]
Thanks @josephlr! Now the wording is a lot better and conveys more clearly what is needed. Even the original text I wrote is not there anymore I learned from this experience :+1:

File: src/lib.rs [hunk 1]
```diff
@@ -74,6 +74,20 @@
...
+//! [dependencies]
+//! getrandom = { version = "0.2", features = ["js"] }
+//! ```
+//!
+//! This can be done even if `getrandom` is not a direct dependency. Cargo
+//! allows crates to enable features for indirect dependencies.
+//!
+//! This feature should only be enabled for binary, test, or benchmark crates.
+//! Library crates should generally not enable this feature, leaving such a
+//! decision to *users* of their library.
```
Comment (newpavlov) [inline src/lib.rs:89]: It may be worth to also add that library crates should not introduce their own `js` features which simply enable `getrandom`'s one.

**Review by newpavlov — APPROVED** [review]

File: src/lib.rs (hunk 1 above)
Comment (josephlr) [reply src/lib.rs]: Good idea, done!

**Pull request merged**